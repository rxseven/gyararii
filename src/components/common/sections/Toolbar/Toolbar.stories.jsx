import React from 'react';
import { storiesOf } from '@storybook/react';

import notes from 'stories/shared/notes';
import Toolbar from './index';

const title = 'Components/Common/Sections/Toolbar';

storiesOf(title, module)
  .add('default', () => <Toolbar>Content</Toolbar>, {
    notes: notes.minimal
  })
  .add('loading', () => <Toolbar isLoading>Content</Toolbar>, {
    notes:
      'A component is invisible while a network request is being submitted.'
  });
