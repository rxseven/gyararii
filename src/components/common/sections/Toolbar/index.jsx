import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';
import styled from 'styled-components';

import Grid from 'components/common/base/Grid';

const Content = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
`;

const Frame = styled.div`
  background-color: #fff;
  display: ${({ isLoading }) => (isLoading ? 'none' : 'block')};
  left: 0;
  margin-top: 0.75rem;
  padding-bottom: 0.25rem;
  padding-top: 46px;
  position: fixed;
  right: 0;
  top: 0;
  width: 100%;
  z-index: 1;
`;

const Shadow = styled.div`
  background: linear-gradient(
    to bottom,
    rgba(255, 255, 255, 1) 0%,
    rgba(255, 255, 255, 0) 100%
  );
  filter: progid:DXImageTransform.Microsoft.gradient(
      startColorstr="#ffffff",
      endColorstr="#00ffffff",
      GradientType=0
    );
  height: 0.5rem;
  left: 0;
  margin-top: 0.25rem;
  position: absolute;
  top: 74px;
  width: 100%;
`;

const propTypes = exact({
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  node: PropTypes.any.isRequired
});

const defaultProps = {
  isLoading: false
};

function Toolbar({ children, isLoading, node }) {
  return (
    <Frame isLoading={isLoading} ref={node}>
      <Grid.Container>
        <Grid.Row>
          <Grid.Column>
            <Content>{children}</Content>
          </Grid.Column>
        </Grid.Row>
        <Shadow />
      </Grid.Container>
    </Frame>
  );
}

Toolbar.propTypes = propTypes;
Toolbar.defaultProps = defaultProps;

export default Toolbar;
