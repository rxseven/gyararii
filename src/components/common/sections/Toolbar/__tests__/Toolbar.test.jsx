import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Toolbar from '../index';

describe('<Toolbar />', () => {
  // Arrange
  const props = { children: 'content' };
  const component = <Toolbar {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
