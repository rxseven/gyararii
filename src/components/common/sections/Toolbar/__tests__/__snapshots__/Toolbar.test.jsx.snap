// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> Snapshot tests should render correctly 1`] = `
<div
  className="sc-bwzfXH hknwyC"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="sc-bdVaJa hBESVI"
        >
          content
        </div>
      </div>
    </div>
    <div
      className="sc-htpNat chXZuS"
    />
  </div>
</div>
`;
